src/application_config/configuring_base.py

```diff
@@ -80,16 +82,35 @@ def _create_instance(cls: type[TConfig]) -> TConfig:
         return cls(**sections)
 
     @classmethod
-    def _get_stored_config(cls) -> dict[str, Any]:
+    def _get_stored_config(cls, configfile_path: str | Path) -> dict[str, Any]:
         """Get the config stored in the toml file"""
         config_stored: dict[str, Any] = {}
-        path = cls.get_configfile_path()
-        try:
-            with path.open(mode="rb") as fptr:
-                config_stored = tomllib.load(fptr)
-        except FileNotFoundError:
+
+        path: Path | None = None
+        if isinstance(configfile_path, Path):
+            path = configfile_path
+        elif configfile_path:
+            if is_valid_filepath(configfile_path, platform="auto"):
+                path = Path(configfile_path)
+            else:
+                raise ValueError
```

Comment:
We should give at least some hint as to what went wrong. Probably something like:

```
ValueError(f"Given path: '{configfile_path}' is not a valid path for this OS")
```

Reply:
Not even sure if this raises a valid ValueError? I would expect you would need to construct it? Are we now not raising the class instance, instead of an actual instantiated value error.